contracts/exchangeIssuance/ExchangeIssuanceZeroEx.sol

```diff
@@ -0,0 +1,561 @@
+            // Check that the component does not have external positions
+            require(
+                _setToken.getExternalPositionModules(components[i]).length == 0,
+                "ExchangeIssuance: EXTERNAL_POSITIONS_NOT_ALLOWED"
+            );
+            approveToken(IERC20(components[i]), _issuanceModule);
+        }
+    }
+
+    /**
+    * Issues an exact amount of SetTokens for given amount of input ERC20 tokens.
+    * The excess amount of tokens is returned in an equivalent amount of ether.
+    *
+    * @param _setToken              Address of the SetToken to be issued
+    * @param _inputToken            Address of the input token
+    * @param _amountSetToken        Amount of SetTokens to issue
+    * @param _maxAmountInputToken   Amount of SetTokens to issue
+    * @param _componentQuotes       The encoded 0x transactions to execute 
+    *
+    * @return totalInputTokenSold   Amount of input token spent for issuance
+    */
+    function issueExactSetFromToken(
+        ISetToken _setToken,
+        IERC20 _inputToken,
+        uint256 _amountSetToken,
+        uint256 _maxAmountInputToken,
+        bytes[] memory _componentQuotes,
+        address _issuanceModule
+    )
+        isWhitelistedIssuanceModule(_issuanceModule)
+        external
+        nonReentrant
+        returns (uint256)
+    {
+
+        _inputToken.safeTransferFrom(msg.sender, address(this), _maxAmountInputToken);
+        _safeApprove(_inputToken, swapTarget, _maxAmountInputToken);
+
+        uint256 totalInputTokenSold = _buyComponentsForInputToken(_setToken, _amountSetToken,  _componentQuotes, _inputToken, _issuanceModule);
+        require(totalInputTokenSold <= _maxAmountInputToken, "ExchangeIssuance: OVERSPENT TOKEN");
+
+        IBasicIssuanceModule(_issuanceModule).issue(_setToken, _amountSetToken, msg.sender);
+
+        _returnExcessInputToken(_inputToken, _maxAmountInputToken, totalInputTokenSold);
+
+        emit ExchangeIssue(msg.sender, _setToken, _inputToken, _maxAmountInputToken, _amountSetToken);
+        return totalInputTokenSold;
+    }
+
+
+    /**
+    * Issues an exact amount of SetTokens for given amount of ETH.
+    * The excess amount of tokens is returned in an equivalent amount of ether.
+    *
+    * @param _setToken              Address of the SetToken to be issued
+    * @param _amountSetToken        Amount of SetTokens to issue
+    * @param _componentQuotes       The encoded 0x transactions to execute
+    *
+    * @return amountEthReturn       Amount of ether returned to the caller
+    */
+    function issueExactSetFromETH(
+        ISetToken _setToken,
+        uint256 _amountSetToken,
+        bytes[] memory _componentQuotes,
+        address _issuanceModule
+    )
+        isWhitelistedIssuanceModule(_issuanceModule)
+        external
+        nonReentrant
+        payable
+        returns (uint256)
+    {
+        require(msg.value > 0, "ExchangeIssuance: INVALID ETH AMOUNT");
```

Comment:
Nit: Could this be clarified to `...: NO ETH SENT` since that's what is checked here?

Reply:
Absolutely. Changed accordingly